### ISSUE TITLE
groupedProfilePlot - error creating separators when number of groups=1

### ISSUE THREAD

**Issue by brownag** [ncss-tech/aqp#50]
```
groupedProfilePlot(p,groups = 'taxonname', label='pedon_id', break.style="line",
                    print.id = TRUE, id.style = 'side', cex.id=1.2, 
                    cex.names=1, cex.depth.axis=1.25, y.offset=7, 
                    axis.line.offset=-3.0, group.name.cex=1, color=input$thematic_field,
                    width=0.1, shrink=T, shrink.cutoff=3)
```
```
guessing horizon designations are stored in `hzname`
Error in segments(y0 = upper.position, y1 = lower.position, x0 = boundary.positions,  : 
  cannot mix zero-length and non-zero-length coordinates
```

**Comment by brownag** [comment]
When running groupedProfilePlot() with `groups` set such that only a single group is returned, an error occurs when trying to add the line/arrow separators. The variable `boundary.positions` has zero-length. This seems to be a new problem that arose since switching over to `segments()` from `abline()` circa 3adfb1.

`boundary.positions <- cumsum(group.lengths)[-length(group.lengths)] + 0.5`

I think it will be simple enough to catch the case where `boundary.positions` has zero length and simply not use group separators. Clearly this isn't a problem when working with SPCs that will result in multiple groups, as groupedProfilePlot was intended. But the issue arises when the plots are generated dynamically from arbitrary SPCs.

**Comment by brownag** [comment]
For the record, the plot output is still generated just fine when running from the console. But the error condition prevents the plot from displaying in the Shiny environment

**Comment by brownag** [comment]
Fixed #51

**Comment by dylanbeaudette** [comment]
Thanks for fixing that.
